lib/kcl.js

```diff
@@ -221,7 +221,21 @@ module.exports = function(config, kinesis) {
 
       if (resp.data.Records && resp.data.Records.length > 0) {
         shard.sequenceNumber = resp.data.Records[resp.data.Records.length - 1].SequenceNumber;
-        config.processRecords.call(shard, resp.data.Records, processRecordsDone);
+        config.processRecords.call(shard, resp.data.Records, function(err, checkpointShard) {
+          if (err) throw err;
+
+          if (checkpointShard) {
+            // save checkpoint to dynamo.
+            db.checkpoint(shard, checkpointSaved);
+          } else {
+            var diff = +new Date() - shard.lastGetRecords;
+            var delay = 0;
+            // if operation takes less than minProcessTime, slow it down to minProcessTime
+            if (diff < config.minProcessTime) delay = config.minProcessTime - diff;
+            logger.info('Processing records took ' + diff + ' ms, delaying ' + delay + ' ms');
```

Comment:
Since kine is a module lets keep logging behind a verbose config option, like https://github.com/mapbox/kine/blob/master/lib/kcl.js#L156